File: dotNet/test/Wall/FakeWallDAO.cs
using System.Collections.Immutable;
using src.Wall;

namespace test.Wall;

public class FakeWallDAO : IWallDAO
{
    private readonly List<Brick> _wall = new();

    public ImmutableArray<Brick> GetBricks(src.User.User user)
    {
        return _wall.ToImmutableArray();
    }

    public void AddBrick(src.User.User user, Brick brick)
    {
        _wall.Add(brick);
    }
}

File: typescript/test/Wall/FakeWallDAO.ts
import {WallDAOInterface} from '../../src/Wall/WallDAOInterface';
import {User} from '../../src/User/User';
import {Brick} from '../../src/Wall/Brick';

export class FakeWallDAO implements WallDAOInterface {
    private bricks: Brick[] = [];

    getBricks(user: User): ReadonlyArray<Brick> {
        return [...this.bricks];
    }

    addBrick(user: User, brick: Brick): void {
        this.bricks.push(brick);
    }
}
